src/axolotl/utils/data.py

```diff
@@ -256,6 +261,17 @@ def for_d_in_datasets(dataset_configs):
                 # https://gcsfs.readthedocs.io/en/latest/#credentials
                 storage_options = {"token": None}
                 remote_file_system = gcsfs.GCSFileSystem(**storage_options)
+            elif config_dataset.path.startswith("puree://"):
+                try:
+                    import gcsfs  # type: ignore
+                except ImportError as exc:
+                    raise ImportError(
+                        "puree:// paths require gcsfs to be installed"
+                    ) from exc
+
+                storage_options = {"token": None}
```

Comment:
is this line the custom logic that we need `puree://` instead of just using `gs://`?

Reply:
It's the other one (line 355-366), this part detects the usage of puree schema. And since puree use GCS, I copied some logic from gs schema.